### PR TITLE
Added ASF Source Headers

### PR DESCRIPTION
**Pull request by smarella**
The policy here seems to indicate even the documentation files that are part of the release tar ball needs to include the source headers, either as a header or as footer.
http://www.apache.org/legal/src-headers.html#header-text

Included it in the footer of the .md files.

### CODE REVIEW

**Comment by smarella** [comment]
Please pardon having several commits as part of the PR. I happened to change most of the .md files on github itself. Will squash them into one commit when merging them.

**Comment by smarella** [comment]
Anyone for a quick review of this? Want to merge this in and do a RC2 asap.

File: gradlew.bat [hunk 1]
```diff
@@ -1,90 +1,106 @@
...
-rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of
-rem the _cmd.exe /c_ return code!
-if  not "" == "%GRADLE_EXIT_CONSOLE%" exit 1
-exit /b 1
-
-:mainEnd
-if "%OS%"=="Windows_NT" endlocal
-
-:omega
+rem Licensed to the Apache Software Foundation (ASF) under one
```
Comment (sdaingade) [inline gradlew.bat]: Does this need @rem ?

Comment (smarella) [inline gradlew.bat]: The Apache RAT tool automatically added this. Looks like the difference between rem and @rem is that @rem prevents the remark to be echoed to stdout when executing the script. But both rem and @rem can be used to mean "comment". 
http://www.computerhope.com/issues/ch001313.htm

**Comment by smarella** [comment]
@klucar has given a LGTM on IRC: http://wilderness.apache.org/channels/?f=myriad/2015-11-12#1447360359

Will merge this in and cut a RC.

**Pull request closed without merging**